src/flamenco/txn/fd_txn_generate.h

```diff
@@ -0,0 +1,87 @@
+/**
+ * Provides utility methods to create txn templates for
+ * pre-staging, as well as a mechanism to build out an 
+ * entire transaction with instructions.
+*/
+
+#include "../../ballet/txn/fd_txn.h"
+#include "../../ballet/txn/fd_compact_u16.h"
+#include "../../flamenco/types/fd_types_custom.h"
+#include "../../flamenco/types/fd_types.h"
+
+#define SIGNATURE_SZ 64
+#define ACCOUNT_SZ   32
+#define BLOCKHASH_SZ 32
+
+/**
+ * Struct used to define a list of accounts supplied in a txn.
+ * Also provides information on number of signers/writeable accounts.
+*/
+struct fd_txn_accounts {
+  ulong signature_cnt;
+  ulong readonly_signed_cnt;
+  ulong readonly_unsigned_cnt;
+  ulong acct_cnt;
+  fd_pubkey_t * signers_w;
+  fd_pubkey_t * signers_r;
+  fd_pubkey_t * non_signers_w;
+  fd_pubkey_t * non_signers_r;
+};
+
+typedef struct fd_txn_accounts fd_txn_accounts_t;
+
+// Message header type
+struct __attribute__((packed)) fd_txn_message_hdr {
+  uchar num_signatures;
+  uchar num_readonly_signatures;
+  uchar num_readonly_unsigned;
+};
+
+typedef struct fd_txn_message_hdr fd_txn_message_hdr_t;
```

Comment:
This isn't meant for use by the callers of this API, right? You can move the definition into the .c file